src/sinks/tap.rs

```diff
@@ -0,0 +1,170 @@
+use super::util::StreamSink;
+use crate::{
+    buffers::Acker,
+    config::{DataType, SinkContext, SinkOuter},
+};
+use crate::{
+    config::SinkConfig,
+    event::{Event, LogEvent},
+};
+use futures::{future, stream::BoxStream, FutureExt, StreamExt};
+use parking_lot::RwLock;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use tokio::sync::broadcast;
+
+type Sender = broadcast::Sender<LogEvent>;
+type Receiver = broadcast::Receiver<LogEvent>;
+
+#[derive(Debug)]
+pub struct TapSink {
+    tx: Sender,
+    acker: Acker,
+}
+
+/// Tap sink is used to 'tap' into events received by uptream components, and broadcast
+/// them to subscribers. Typically, this is used to expose events to the API, but is general
+/// purpose enought that it could technically be used with other mechanisms. This sink is
+/// not added to inventory; it's not intended to be user configurable.
```

Comment:
```suggestion
/// Tap sink is used to 'tap' into events received by upstream components, and broadcast
/// them to subscribers. Typically, this is used to expose events to the API, but is general
/// purpose enough that it could technically be used with other mechanisms. This sink is
/// not added to inventory; it's not intended to be user configurable.
```